### PR TITLE
MF-22: Switching to using openmrsFetch

### PR DESCRIPTION
**Pull request by joeldenning**
See related https://issues.openmrs.org/projects/MF/issues/MF-22?filter=allopenissues

### CODE REVIEW

File: src/login/login.resource.tsx [hunk 1]
```diff
@@ -1,20 +1,10 @@
...
-    .then(resp => {
-      if (resp.ok) {
-        return resp.json();
-      } else {
-        throw Error(
-          `Could not login. Server responded with status ${resp.status}`
-        );
-      }
-    });
+  return openmrsFetch(`/ws/rest/v1/session`, {
```
Comment (joeldenning) [inline src/login/login.resource.tsx]: Notice how `/openmrs` is no longer needed

File: src/login/login.resource.tsx [hunk 2]
```diff
@@ -1,20 +1,10 @@
+import { openmrsFetch } from "@openmrs/esm-api";
+
 export function performLogin(username, password) {
   const token = window.btoa(`${username}:${password}`);
-  return window
-    .fetch(`/openmrs/ws/rest/v1/session`, {
-      method: "GET",
-      headers: {
-        Accept: "application/json",
```
Comment (joeldenning) [inline src/login/login.resource.tsx]: Notice how `/Accept` is no longer needed -- done automatically in `@openmrs/esm-api`

File: src/login/login.resource.tsx [hunk 3]
```diff
@@ -1,20 +1,10 @@
...
-        Authorization: `Basic ${token}`
-      }
-    })
-    .then(resp => {
-      if (resp.ok) {
-        return resp.json();
```
Comment (joeldenning) [inline src/login/login.resource.tsx]: Notice how checking the response status or downloading the json is no longer needed -- done automatically in `@openmrs/esm-api`

**Comment by jonathandick** [comment]
Definitely makes things easier.

**Pull request merged**